dns/test/zones.test.js

```diff
@@ -0,0 +1,37 @@
+/**
+ * Copyright 2016, Google, Inc.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const proxyquire = require(`proxyquire`).noCallThru();
+
+describe(`dns:zones`, () => {
+  it(`should handle errors`, () => {
+    const error = new Error(`error`);
+    const callback = sinon.spy();
+    const dnsMock = {
+      getZones: sinon.stub().yields(error)
+    };
+    const DNSMock = sinon.stub().returns(dnsMock);
+    const program = proxyquire(`../zones`, {
+      '@google-cloud/dns': DNSMock
+    });
+
+    program.listZones(callback);
+
+    assert.equal(callback.callCount, 1);
```

Comment:
tldr; sinon only has `calledOnce`, `calledTwice`, and `calledThrice`. If the spy is called more than that, you have to use `callCount`. To create this test I copied another test that called the spy 8 times.